webservices/cxf/src/java/org/sakaiproject/webservices/SakaiScript.java

```diff
@@ -4243,25 +4278,179 @@ public String copySiteContentForTool(
      * @param fromContext The context to import from.
      * @param toContext   The context to import into.
      */
-    private void transferCopyEntities(String toolId, String fromContext, String toContext) {
-        // offer to all EntityProducers
-        for (Iterator i = entityManager.getEntityProducers().iterator(); i.hasNext(); ) {
-            EntityProducer ep = (EntityProducer) i.next();
-            if (ep instanceof EntityTransferrer) {
-                try {
-                    EntityTransferrer et = (EntityTransferrer) ep;
-
-                    // if this producer claims this tool id
-                    if (ArrayUtil.contains(et.myToolIds(), toolId)) {
-                        et.transferCopyEntities(fromContext, toContext, new ArrayList());
-                    }
-                } catch (Throwable t) {
-                    LOG.warn(this + ".transferCopyEntities: Error encountered while asking EntityTransfer to transferCopyEntities from: " + fromContext + " to: " + toContext, t);
-                }
-            }
-        }
+    protected Map transferCopyEntities(String toolId, String fromContext, String toContext)
+    {
+    	Map transversalMap = new HashMap();
+
+    	// offer to all EntityProducers
+    	for (Iterator i = entityManager.getEntityProducers().iterator(); i.hasNext();)
+    	{
+    		EntityProducer ep = (EntityProducer) i.next();
+    		if (ep instanceof EntityTransferrer)
+    		{
+    			try
+    			{
+    				EntityTransferrer et = (EntityTransferrer) ep;
+
+    				// if this producer claims this tool id
+    				if (ArrayUtil.contains(et.myToolIds(), toolId))
+    				{
+    					if(ep instanceof EntityTransferrerRefMigrator)
+    					{
+    						EntityTransferrerRefMigrator etMp = (EntityTransferrerRefMigrator) ep;
+    						Map<String,String> entityMap = etMp.transferCopyEntitiesRefMigrator(fromContext, toContext, new ArrayList(), true);
+    						if(entityMap != null)
+    						{
+    							transversalMap.putAll(entityMap);
+    						}
+    					}
+    					else
+    					{
+    						et.transferCopyEntities(fromContext, toContext,	new ArrayList(), true);
+    					}
+    				}
+    			}
+    			catch (Throwable t)
+    			{
+    				LOG.warn(this + ".transferCopyEntities: Error encountered while asking EntityTransfer to transferCopyEntities from: " + fromContext + " to: " + toContext, t);
```

Comment:
You can shorten the log messages as they will output the class and method so there is no need to include them in the log message.
`LOG.warn("Error encountered while asking EntityTransfer to transferCopyEntities from: " + fromContext + " to: " + toContext, t)`